pkg/highlight/highlighter.go

```diff
@@ -454,7 +454,8 @@ func (h *Highlighter) HighlightString(input string) []LineMatch {
 	return lineMatches
 }
 
-// Highlight sets the state and matches for each line from startline to endline
+// Highlight sets the state and matches for each line beginning with the startline
+// until it detects an unchanged state from the endline downwards.
```

Comment:
nit: As we are already cleaning up this description, I'd try and make it clear and explicit right away, something like "...for each line from startline to endline, and also some amount of lines after endline, until it detects a line whose state does not change, which means that the lines after it do not change their highlighting and therefore do not need to be updated."

Also, my previous comment was actually about the sentence "It sets all other matches in the buffer to nil to conserve memory.". Looks like we should just remove it?